hoot-core/src/main/cpp/hoot/core/visitors/AddRef1Visitor.cpp

```diff
@@ -28,28 +28,25 @@
 
 // hoot
 #include <hoot/core/util/Factory.h>
-#include <hoot/core/elements/OsmMap.h>
 #include <hoot/core/util/ConfigOptions.h>
 #include <hoot/core/schema/MetadataTags.h>
 
 namespace hoot
 {
 
-HOOT_FACTORY_REGISTER(ConstElementVisitor, AddRef1Visitor)
+HOOT_FACTORY_REGISTER(ElementVisitor, AddRef1Visitor)
 
 AddRef1Visitor::AddRef1Visitor()
 {
   _count = 0;
   setConfiguration(conf());
 }
 
-void AddRef1Visitor::visit(const ConstElementPtr& e)
+void AddRef1Visitor::visit(const ElementPtr& e)
 {
-  boost::shared_ptr<Element> ee = _map->getElement(e->getElementId());
-
-  if (_informationOnly == false || ee->getTags().getNonDebugCount() > 0)
+  if (_informationOnly == false || e->getTags().getNonDebugCount() > 0)
   {
-    ee->getTags()[MetadataTags::Ref1()] = _prefix + QString("%1").arg(_count++, 6, 16, QChar('0'));
+    e->getTags()[MetadataTags::Ref1()] = _prefix + QString("%1").arg(_count++, 6, 16, QChar('0'));
   }
```

Comment:
No map consumption needed here, and this is actually an ElementVisitor, since its modifying an element.  To use it with FilteredVisitor (if needed), though, at this point would require inheriting ElementOsmMapVisitor instead.